File: Javascript/algorithms/sorting/InsertionSort.js
const InsertionSort = (arr) => {
  for (let i = 1; i < arr.length; i++) {
    let currentValue = arr[i]
    for (let j = i - 1; j >= 0 && arr[j] > currentValue; j--) {
      console.log(arr)
      arr[j + 1] = arr[j]
      arr[j] = currentValue
    }
  }
  return arr
}

// console.log(InsertionSort([2, 1, 9, 76, 4]))
console.log(InsertionSort([12, 11, 13, 5, 6]))

File: Java/algorithms/sorting/InsertionSort.java
package Java.algorithms.sorting;

import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {5,3,7,8,10,12};
        int[] result = insertionSort(arr);
        System.out.println(Arrays.toString(result));
    }

    public static int[] insertionSort(int[] arr){
        for(int i = 1; i < arr.length ; i++){
            int current = arr[i];
            for(int j = i-1 ; j >= 0 && arr[j] > current ; j--){
                arr[j+1] = arr[j];
                arr[j] = current;
            }
        }
        return arr;
    }
}
